utils/v2/connector/impl/RedStonePriceFeed.ts

```diff
@@ -1,47 +1,116 @@
-import {PriceDataType, PriceFeedConnector, SignedPriceDataType} from "../PriceFeedConnector";
+import { PriceFeedConnector, SignedPriceDataType } from "../PriceFeedConnector";
 import axios from "axios";
 import _ from "lodash";
-import EvmPriceSigner from "redstone-node/dist/src/signers/EvmPriceSigner";
+import { Fetcher, SignedDataPackageResponse, SourceConfig } from "./fetchers/Fetcher";
+import { createFetcher } from "./fetchers";
+import { convertResponseToPricePackage, selectDataPackage, validateDataPackage } from "./DataPackageUtils";
 
-export type RedStoneProvider = "redstone" | "redstone-stocks" | "redstone-rapid";
+const DEFAULT_TIMEOUT_MILLISECONDS = 10000; // 10 seconds
+
+export type ValueSelectionAlgorithm = "first-valid" | "newest-valid" | "oldest-valid"; // wa can add "median-valid" in future
+export type SourceType = "cache-layer" | "streamr" | "streamr-storage";
+
+export interface DataSourcesConfig {
+  valueSelectionAlgorithm: ValueSelectionAlgorithm,
+  timeoutMilliseconds: number;
+  maxTimestampDiffMilliseconds: number;
+  preVerifySignatureOffchain: boolean;
+  signers?: string[], // Will be used in future (after multi-nodes creation)
+  sources: SourceConfig[],
+};
+
+export interface PriceFeedOptions {
+  dataSources?: DataSourcesConfig;
+  asset?: string;
+};
+
+export type RedStoneProvider =
```

Comment:
Do you think it's good to strong-type it here, wouldn't it be harder to add new providers?

Reply:
I would keep it this way for now, as we actually want to motivate people to collaborate with us while using oracles. Because we want to avoid situations, when someone uses our oracles (in a wrong way) and tells that redstone doesn't work

Reply:
Fine, but could we keep somewhere a document what's necessary to add a new provider, so we won't forget that we should update the evm-connector code.

Reply:
Sure! However I would not forget about it, because I would do an end to end testing each time when we add a new provider :)